x-pack/plugins/infra/public/pages/logs/settings/source_configuration_form_state.tsx

```diff
@@ -5,103 +5,69 @@
  * 2.0.
  */
 
-import { useCallback, useMemo } from 'react';
-import { ResolvedLogSourceConfiguration } from '../../../../common/log_sources';
-import { useLogIndicesConfigurationFormState } from './indices_configuration_form_state';
-import { useLogColumnsConfigurationFormState } from './log_columns_configuration_form_state';
+import { useMemo } from 'react';
+import { LogSourceConfigurationProperties } from '../../../containers/logs/log_source';
+import { useCompositeFormElement } from './form_elements';
+import { useFieldsFormElement, useLogIndicesFormElement } from './indices_configuration_form_state';
+import { useLogColumnsFormElement } from './log_columns_configuration_form_state';
+import { useNameFormElement } from './name_configuration_form_state';
 
 export const useLogSourceConfigurationFormState = (
-  configuration?: ResolvedLogSourceConfiguration
+  configuration?: LogSourceConfigurationProperties
 ) => {
-  const indicesConfigurationFormState = useLogIndicesConfigurationFormState({
-    initialFormState: useMemo(
-      () =>
-        configuration
-          ? {
-              name: configuration.name,
-              description: configuration.description,
-              logAlias: configuration.indices,
-              tiebreakerField: configuration.tiebreakerField,
-              timestampField: configuration.timestampField,
-            }
-          : undefined,
-      [configuration]
-    ),
-  });
+  const nameFormElement = useNameFormElement(configuration?.name ?? '');
 
-  const logColumnsConfigurationFormState = useLogColumnsConfigurationFormState({
-    initialFormState: useMemo(
+  const logIndicesFormElement = useLogIndicesFormElement(
+    useMemo(
       () =>
-        configuration
-          ? {
-              logColumns: configuration.columns,
-            }
-          : undefined,
+        configuration?.logIndices ?? {
```

Comment:
Thinking aloud here really: should we default to something here? The configuration may be `undefined` at the top level when it hasn't loaded etc, but once loaded (successfully, without an error) the expectation is that it always has defaults set (otherwise something is broken in our server side handling). Is it misleading to have a further fallback here for the initial value? Rather than relying on blocking access to the form until we have the configuration?

Reply:
Ideally, yes. But the type checker doesn't know that it will not be rendered with `undefined`. And we sometimes need that capability to work around the laws of hooks not allowing conditional calls to hooks.

I think we hide the form for the user while it's loading already. Or am I misunderstanding your question?